### DISCUSSION TITLE
Merge Strategies and Dev Branches

### DISCUSSION THREAD

**Discussion by sakethramanujam** [GVP-AI-Club/gvp-ai-club#15] [Ideas]
So, for the sanity of the repository I would like to suggest that there be feature branches that get feature updates from individual developer branches. They then get merged into the main/deployable branches after proper code review.

There was a discussion about having a contributors guidelines for org wide repos.
If you folks find a good contributing.md please post a link it here and thus you can vote on having one for the org and follow it.

**Comment by rakesh0x7** [comment]
@sakethramanujam  we are following [this](https://github.com/gdsc-gvp/DSA/blob/main/.github/CONTRIBUTING.md) for GDSC Org and  Could you please let me know if it is a good fit for the AI club organization, or if there is anything we can do to improve it 😄

**Reply by sakethramanujam** [reply]
it's good to start with.

Here are some additional comments.

1. Add Issue Templates (Not Urgent, but look into it)
2. Features get development branches preferably a naming structure would be `repo/develop/<issue-number/name>/<feature>/<feature-name>`
3. Nothing gets merged onto a master branch from feature branches directly. It goes to a staging branch, which can follow `/repo/staging/<feature/issue>` and it gets a test deployment or stays there until there are no traceable bugs present.
4. The commit(s) then get merged into master, usually one commit per feature until unless it requires more than one commit. 
5. No merge commits please.
6. Every branch needs to be regularly rebased to reflect HEAD->master state within the branch.


Please let me know if you have any issues in understanding this.
